haystack/utils/deepsetcloud.py

```diff
@@ -635,6 +637,115 @@ def _build_workspace_url(self, workspace: Optional[str] = None):
+        except IndexError:
+            raise DeepsetCloudError(f"No evaluation set found with the name {evaluation_set_name}")
+
+        labels = self._get_labels_from_evaluation_set(
+            workspace=workspace, evaluation_set_id=evaluation_set["evaluation_set_id"]
+        )
+
+        return [
+            Label(
+                query=label_dict["query"],
+                document=Document(content=label_dict["context"]),
+                is_correct_answer=True,
+                is_correct_document=True,
+                origin="user-feedback",
+                answer=Answer(label_dict["answer"]),
+                id=label_dict["label_id"],
+                no_answer=False if label_dict.get("answer", None) else True,
+                pipeline_id=None,
+                created_at=None,
+                updated_at=None,
+                meta=label_dict["meta"],
+                filters={},
+            )
+            for label_dict in labels
+        ]
+
+    def get_labels_count(self, evaluation_set_name: Optional[str] = None, workspace: Optional[str] = None) -> int:
+        """
+        Counts labels for a given evaluation set in deepset cloud.
+
+        :param evaluation_set_name: Optional index in Deepset Cloud
+                      If None, the EvaluationSetClient's default index (self.index) will be used.
+        :param workspace: Optional workspace in Deepset Cloud
+                          If None, the EvaluationSetClient's default workspace (self.workspace) will be used.
+
+        :return: Number of labels for the given (or defaulting) index
+        """
+        evaluation_set = self._get_evaluation_set(evaluation_set_name=evaluation_set_name, workspace=workspace)
+        return evaluation_set[0]["total_labels"]
+
+    def get_evaluation_set_names(self, workspace: Optional[str] = None):
+        """
+        Searches for all evaluation set names in the given workspace in Deepset Cloud.
+
+        :param workspace: Optional workspace in Deepset Cloud
+                          If None, the EvaluationSetClient's default workspace (self.workspace) will be used.
+
+        :return: list of Label
+        """
+        evaluation_sets_response = self._get_evaluation_set(evaluation_set_name=None, workspace=workspace)
+
+        return [eval_set["name"] for eval_set in evaluation_sets_response]
+
+    def _get_evaluation_set(self, evaluation_set_name: Optional[str], workspace: Optional[str] = None) -> List[dict]:
+        if not evaluation_set_name:
+            evaluation_set_name = self.evaluation_set_name
+
+        url = self._build_workspace_url(workspace=workspace)
+        evaluation_set_url = f"{url}/evaluation_sets"
+
+        for response in self.client.get_with_auto_paging(
+            url=evaluation_set_url, query_params={"name": evaluation_set_name}
+        ):
+            return response.json().get("data", [])
```

Comment:
That doesn't seem to fit together. `get_with_auto_paging` returns a generator of the objects within the "data" property.

Reply:
Ahh true, let me change that :)

Reply:
Ahh true, let me change that :)